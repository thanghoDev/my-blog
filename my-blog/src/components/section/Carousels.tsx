import Image from 'next/future/image';
import Link from 'next/link';
import { Carousel } from 'react-bootstrap';

// constant
import { CATEGORY, DETAIL } from 'constant/Pages';

// types
import { Blog } from '@/types/blog';
type CarouselsProps = {
  carousel: Blog[];
};

function Carousels({ carousel }: CarouselsProps) {
  return (
    <Carousel variant='dark' className='mt-5 d-flex bg-light'>
      {carousel.map((item) => (
        <Carousel.Item key={item.id}>
          <Image
            className='d-block'
            width={555}
            height={377}
            src={item.images}
            alt={item.alt}
          />
          <Carousel.Caption className='text-dark text-start w-50 top-0 start-50 ps-4'>
            <Link href={`${DETAIL}/${item.id}`}>
              <h3 className='text-capitalize text-black postTitle'>
                <a>{item.title}</a>
              </h3>
            </Link>
            <p className='description'>{item.description}</p>
            <p className='text-capitalize category mt-3'>
              Dave Rogers <code className='text-secondary'>in</code>{' '}
              <Link href={`/${CATEGORY}/${item.category}`}>
                <a className='text-black text-decoration-none'>
                  {item.category}
                </a>
              </Link>
            </p>
          </Carousel.Caption>
        </Carousel.Item>
      ))}
    </Carousel>
  );
}

export default Carousels;
