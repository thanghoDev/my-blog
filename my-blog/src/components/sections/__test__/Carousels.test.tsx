import React from 'react';
import { render, screen } from '@testing-library/react';

import Carousels from '../Carousels';

describe('testing Carousels', () => {
  const carousel = [
    {
      id: 1,
      title: 'title',
      description: 'description',
      category: 'carousel',
      view: 150,
      images: 'https://i.ibb.co/10hj3h1/carousel.jpg',
      alt: 'carousel',
      content: 'content',
    },
  ];
  test('render Carousels', () => {
    const { container } = render(<Carousels carousel={carousel} />);
    expect(container.firstChild).toMatchSnapshot();

    const title = screen.getByText(/title/i);
    expect(title).toBeInTheDocument();

    const image = screen.getByRole('img', {
      name: /carousel/i,
    });
    expect(image).toBeInTheDocument();
  });
});
