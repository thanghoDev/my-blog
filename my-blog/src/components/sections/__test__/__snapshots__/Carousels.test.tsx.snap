// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Carousels render Carousels 1`] = `
<div
  class="mt-5 d-flex bg-light carousel slide carousel-dark"
>
  <div
    class="carousel-indicators"
  >
    <button
      aria-current="true"
      aria-label="Slide 1"
      class="active"
      data-bs-target=""
      type="button"
    />
  </div>
  <div
    class="carousel-inner"
  >
    <div
      class="active carousel-item"
    >
      <img
        alt="carousel"
        class="d-block"
        data-nimg="future"
        decoding="async"
        height="377"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fcarousel.jpg&w=1200&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fcarousel.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fcarousel.jpg&w=1200&q=75 2x"
        style="color: transparent;"
        width="555"
      />
      <div
        class="text-dark text-start w-50 top-0 start-50 ps-4 carousel-caption"
      >
        <h3
          class="text-capitalize text-black postTitle"
        >
          <a>
            title
          </a>
        </h3>
        <p
          class="description"
        >
          description
        </p>
        <p
          class="text-capitalize category mt-3"
        >
          Dave Rogers 
          <code
            class="text-secondary"
          >
            in
          </code>
           
          <a
            class="text-black text-decoration-none"
            href="/category/carousel"
          >
            carousel
          </a>
        </p>
      </div>
    </div>
  </div>
  <a
    class="carousel-control-prev"
    href="#"
    role="button"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="carousel-control-prev-icon"
    />
    <span
      class="visually-hidden"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    href="#"
    role="button"
    tabindex="0"
  >
    <span
      aria-hidden="true"
      class="carousel-control-next-icon"
    />
    <span
      class="visually-hidden"
    >
      Next
    </span>
  </a>
</div>
`;
